frontends/infinite-corridor/src/pages/user-lists/UpsertListDialog.tsx

```diff
@@ -0,0 +1,226 @@
+          Create a list of of any of our learning resources.
+        </span>
+      </>
+    ),
+    className: "radio-option"
+  },
+  {
+    value: LRType.LearningPath,
+    label: (
+      <>
+        <span className="option-header">Learning Path</span>
+        <span className="option-detail">
+          An ordered list of learning resources.
+        </span>
+      </>
+    ),
+    className: "radio-option"
+  }
+]
+
+const PRIVACY_CHOICES = [
+  {
+    value:     PrivacyLevel.Private,
+    label:     "Private",
+    className: "radio-option"
+  },
+  {
+    value:     PrivacyLevel.Public,
+    label:     "Public",
+    className: "radio-option"
+  }
+]
+
+interface UpsertListDialogProps {
+  title: string
+  open: boolean
+  resource?: UserList | null
+  onClose: () => void
+}
+const UpsertListDialog: React.FC<UpsertListDialogProps> = ({
+  resource,
+  open,
+  onClose,
+  title
+}) => {
+  const createUserList = useCreateUserList()
+  const updateUserList = useUpdateUserList()
+  const handleSubmit: FormikConfig<Partial<ListFormSchema>>["onSubmit"] =
+    useCallback(
+      async values => {
+        if (resource?.id) {
+          await updateUserList.mutateAsync({ id: resource.id, ...values })
+        } else {
+          await createUserList.mutateAsync(values)
+        }
+        onClose()
+      },
+      [resource, onClose, createUserList, updateUserList]
+    )
+  const formik = useFormik({
+    enableReinitialize: true,
+    initialValues:
+      resource ?? (listFormSchema.getDefault() as Partial<ListFormSchema>),
+    validationSchema: listFormSchema,
+    onSubmit:         handleSubmit,
+    validateOnChange: false,
+    validateOnBlur:   false
+  })
+
+  const topicsQuery = useTopics()
+  const topics = topicsQuery.data?.results ?? []
+  const hasError = createUserList.isError || updateUserList.isError
+
+  const priaceChoices = PRIVACY_CHOICES.filter(choice => {
+    if (choice.value === PrivacyLevel.Public) return SETTINGS.user.is_list_staff
+    return true
+  })
+
+  return (
+    <FormDialog
+      open={open}
+      title={title}
+      formClassName="manage-list-form"
+      onClose={onClose}
+      onReset={formik.resetForm}
+      onSubmit={formik.handleSubmit}
+      noValidate
+      footerContent={
+        hasError &&
+        !formik.isSubmitting && (
+          <Alert severity="error">
+            There was a problem saving your list. Please try again later.
+          </Alert>
+        )
+      }
+    >
+      <RadioChoiceField
+        className="form-row"
+        name="list_type"
+        label="List Type"
+        choices={LIST_TYPE_CHOICES}
+        value={formik.values.list_type}
+        row
+        onChange={formik.handleChange}
+      />
+      <RadioChoiceField
+        className="form-row"
+        name="privacy_level"
+        label="Privacy"
+        choices={priaceChoices}
+        value={formik.values.privacy_level}
+        row
+        onChange={formik.handleChange}
+      />
```

Comment:
There's a typo in `priaceChoices`, but when I thought on it more, is there even any value to show this option to non-staff because they can only make private lists and they'll only ever have one option? Should we just do something where it's a hidden input with `PRIVATE` if they're non-staff, otherwise they get the radio and you don't need to filter the options?

Reply:
I guess showing it communicates that the list is private, but not much else. Showing nothing seems better so I've hidden it for now. I'll solicit some more feedback from Bilal and Ferdi on the followup PR that has "add to list" functionality.

Here is the current version (for non staff)

<img width="627" alt="Screenshot 2023-03-10 at 2 59 26 PM" src="https://user-images.githubusercontent.com/9010790/224416150-bb0154a6-83b8-44f6-9049-d8680de01052.png">